s3/s3.go

```diff
@@ -829,13 +832,25 @@ func (o *Object) Hash(t fs.HashType) (string, error) {
 	if t != fs.HashMD5 {
 		return "", fs.ErrHashUnsupported
 	}
-	etag := strings.Trim(strings.ToLower(o.etag), `"`)
+	hash := strings.Trim(strings.ToLower(o.etag), `"`)
 	// Check the etag is a valid md5sum
-	if !matchMd5.MatchString(etag) {
-		// fs.Debugf(o, "Invalid md5sum (probably multipart uploaded) - ignoring: %q", etag)
-		return "", nil
+	if !matchMd5.MatchString(hash) {
+		err := o.readMetaData()
+		if err != nil {
+			return "", err
+		}
+
+		if md5sum, ok := o.meta[metaMD5Hash]; ok {
+			md5sumBytes, err := base64.StdEncoding.DecodeString(*md5sum)
```

Comment:
Thinking about this some more, I think it would be best to store the md5 hash in the same format as the Etag, namely as hex digits.  What do you think?

Reply:
Hhmmm... there are a couple of competing consistencies here, so I'm (not 100%) happy either way:

1. My preference is to keep `md5chksum's` encoding consistent with `Content-MD5's` encoding (base64) ; and to consistently follow AWS's advice on both `md5chksum's` name and its encoding.
1. We could instead keep `md5chksum's` encoding consistent with `ETag's` (hex), and also with rclone's encoding of other sums.

I like the first but the second works, too: whichever you like.  A third option would be to follow AWS's advice on neither the name nor the encoding: call the field something else and encode it as hex.